Tmain/utils.sh

```diff
@@ -135,3 +135,10 @@ direq ()
     [ "$(cd ${1} && pwd)" = "$(cd ${2} && pwd)" ]
     return $?
 }
+
+skip_if_grep_doesnt_work_expectedly()
+{
+	if ! ( echo 'MatLab   *.m' | grep '\*\.m\>.*$' > /dev/null ); then
```

Comment:
Ugly.

Reply:
I don't understand why you are jumping through such excessive and unwieldy hoops.

The issue is that POSIX `grep -E` is not the default behavior of POSIX grep, but it is the default for GNU grep. Right? So... why not just explicitly specify -E everywhere?

Reply:
Thank you. It seems that grep -E doesn't work well on FreeBSD 13 though it works well on FreeBSD 12.

Reply:
grep command on FreeBSD13 has -E option.
`grep --version` reports:
```
grep (BSD grep, GNU compatible) 2.6.0-FreeBSD
```
`grep --help` says:
```
usage: grep [-abcDEFGHhIiLlmnOoPqRSsUVvwxz] [-A num] [-B num] [-C[num]]
+	[-e pattern] [-f file] [--binary-files=value] [--color=when]
+	[--context[=num]] [--directories=action] [--label] [--line-buffered]
+	[--null] [pattern] [file ...]
```

```
echo 'MatLab   *.m' | grep -E '\*\.m\>.*$'
```
This pipeline prints nothing.
This is the reason why the failure occurs only on FreeBSD.

The pattern may include a subpattern supported in a modern GNU extension.

Reply:
I collected the version information of grep on FreeBSD12:
```
grep (GNU grep) 2.5.1-FreeBSD
```

It seems that the pattern used in the test sports an incompatible aspect of the compatible implementation.

If you know an alternative pattern for the pattern, a pull request is welcome.
However, this pull request becomes a showstopper in parser developments.
So I will merge the original change.

Reply:
I use -E after testing: 2457f86.